### PR TITLE
DOC Document new dev admin permissions

### PR DESCRIPTION
**Pull request by GuySartorelli**
## Issue
- https://github.com/silverstripe/silverstripe-framework/issues/10852

### CODE REVIEW

File: en/04_Changelogs/5.2.0.md [hunk 1]
```diff
@@ -103,6 +104,14 @@ The files section of the CMS now has buttons to select and deselect all files an
 
 ![](../_images/asset-admin-select-all.png)
 
+### More nuanced permissions for `/dev/*` routes {#dev-route-permissions}
+
+Previously, all `/dev/*` routes registered with [`DevelopmentAdmin`](api:SilverStripe\Dev\DevelopmentAdmin) (for example `/dev/tasks/MyBuildTask`) could only be access by administrator users, and this couldn't be configured.
+
+Now, all of the controllers which handle these routes that come packaged in a core or supported module have configurable permissions that can be used to grant non-administrative users access to these routes.
+
+You can also now optonally implement a `canView()` method on your `BuildTask` implementations to restrict access for specific tasks even further. This means you can grant access to _some_ tasks to specific users or groups without granting access to _all_ tasks.
```
Comment (andrewandante) [inline en/04_Changelogs/5.2.0.md]:
```suggestion
You can also now optionally implement a `canView()` method on your `BuildTask` implementations to restrict access for specific tasks even further. This means you can grant access to _some_ tasks to specific users or groups without granting access to _all_ tasks.
```

File: en/04_Changelogs/5.2.0.md [hunk 2]
```diff
@@ -103,6 +104,14 @@ The files section of the CMS now has buttons to select and deselect all files an
 
 ![](../_images/asset-admin-select-all.png)
 
+### More nuanced permissions for `/dev/*` routes {#dev-route-permissions}
+
+Previously, all `/dev/*` routes registered with [`DevelopmentAdmin`](api:SilverStripe\Dev\DevelopmentAdmin) (for example `/dev/tasks/MyBuildTask`) could only be access by administrator users, and this couldn't be configured.
```
Comment (andrewandante) [inline en/04_Changelogs/5.2.0.md:109]:
```suggestion
Previously, all `/dev/*` routes registered with [`DevelopmentAdmin`](api:SilverStripe\Dev\DevelopmentAdmin) (for example `/dev/tasks/MyBuildTask`) could only be accessed by administrator users, and this couldn't be configured.
```

File: en/04_Changelogs/5.2.0.md [hunk 3]
```diff
@@ -103,6 +104,14 @@ The files section of the CMS now has buttons to select and deselect all files an
...
+
+Now, all of the controllers which handle these routes that come packaged in a core or supported module have configurable permissions that can be used to grant non-administrative users access to these routes.
```
Comment (andrewandante) [inline en/04_Changelogs/5.2.0.md]:
```suggestion
Now, all of the controllers which handle these routes that come packaged in a core or supported module have configurable `$init_permissions` that can be used to grant non-administrative users access to these routes.
```

**Review by andrewandante — CHANGES_REQUESTED** [review]
Couple of typos, I'd add the actual name of the config option because as a dev, that's what I'm scanning for.

I think this page also needs to be updated: https://docs.silverstripe.org/en/5/developer_guides/controllers/builtin_middlewares/

File: en/04_Changelogs/5.2.0.md (hunk 3 above)
Comment (ChloeSartorelli) [reply en/04_Changelogs/5.2.0.md]: Done

File: en/04_Changelogs/5.2.0.md (hunk 2 above)
Comment (ChloeSartorelli) [reply en/04_Changelogs/5.2.0.md:109]: Done

File: en/04_Changelogs/5.2.0.md (hunk 1 above)
Comment (ChloeSartorelli) [reply en/04_Changelogs/5.2.0.md]: Done

**Comment by GuySartorelli** [comment]
> I think this page also needs to be updated: https://docs.silverstripe.org/en/5/developer_guides/controllers/builtin_middlewares/

What specific changes did you have in mind?

**Comment by andrewandante** [comment]
> > I think this page also needs to be updated: https://docs.silverstripe.org/en/5/developer_guides/controllers/builtin_middlewares/
> 
> What specific changes did you have in mind?

Basically adding in a link to `DevelopmentAdminConfirmationMiddleware` along with the other ones, and a short description. Though I now note that it says *some* of them can be found here, so I guess it's up to you 😄

**Review by andrewandante — APPROVED** [review]

**Comment by GuySartorelli** [comment]
Added a row to the table on that page. Also added backticks to all items on that table.

File: en/02_Developer_Guides/02_Controllers/06_Builtin_Middlewares.md [hunk 4]
```diff
@@ -6,20 +6,21 @@ summary: Middleware components that come with Silverstripe CMS
...
-| [URLSpecialsMiddleware](api:SilverStripe\Control\Middleware\URLSpecialsMiddleware) | Controls some of the [URL special variables](../debugging/url_variable_tools) |
+| [`AllowedHostsMiddleware`](api:SilverStripe\Control\Middleware\AllowedHostsMiddleware) | Secures requests by only allowing a whitelist of Host values |
+| [`AuthenticationMiddleware`](api:SilverStripe\Security\AuthenticationMiddleware) | Handles authentication for the request |
+| [`CanonicalURLMiddleware`](api:SilverStripe\Control\Middleware\CanonicalURLMiddleware) | URL normalisation and redirection |
+| [`ChangeDetectionMiddleware`](api:SilverStripe\Control\Middleware\ChangeDetectionMiddleware) | Change detection via Etag / IfModifiedSince headers, conditionally sending a 304 not modified if possible. |
+| [`ConfirmationMiddleware`](api:SilverStripe\Control\Middleware\ConfirmationMiddleware) | Checks whether user manual confirmation is required for HTTPRequest |
+| [`DevelopmentAdminConfirmationMiddleware`](api:SilverStripe\Control\Middleware\DevelopmentAdminConfirmationMiddleware) | A specific subclass of `ConfirmationMiddleware` which handles the `/dev/*` routes. |
```
Comment (ChloeSartorelli) [inline en/02_Developer_Guides/02_Controllers/06_Builtin_Middlewares.md:18]: This `DevelopmentAdminConfirmationMiddleware` row is the only new content on this page. Other than this, the only change was adding backticks.

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**